code/modules/clothing/gloves/miscellaneous.dm

```diff
@@ -12,6 +12,130 @@
+	name = "armwraps of the hungry ghost"
+	desc = "A series of blackened, bloodstained armwraps stitched with strange geometric symbols. Makes you pretty keen to commit horrible acts against the living through bloody carnage."
+	icon_state = "narsiearmwraps"
+	item_state = "narsiearmwraps"
+	resistance_flags = FIRE_PROOF | ACID_PROOF
+	armor = list("melee" = 10, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 35, "rad" = 0, "fire" = 50, "acid" = 50)
+	enhancement = 3
+	secondary_trait = TRAIT_KI_VAMPIRE
+	
+/obj/item/clothing/gloves/fingerless/pugilist/brassmountain
+	name = "armbands of the brass mountain"
+	desc = "A series of scolding hot brass armbands. Makes you pretty keen to bring the light to the unenlightened through unmitigated violence."
+	icon_state = "ratvararmwraps"
+	item_state = "ratvararmwraps"
+	resistance_flags = FIRE_PROOF | ACID_PROOF
+	armor = list("melee" = 10, "bullet" = 0, "laser" = -10, "energy" = 0, "bomb" = 0, "bio" = 35, "rad" = 0, "fire" = 50, "acid" = 50)
+	enhancement = 4 //The artifice of Ratvar is unmatched except when it is.
+	secondary_trait = TRAIT_STRONG_GRABBER
+
+/obj/item/clothing/gloves/fingerless/pugilist/rapid
+	name = "Bands of the North Star"
+	desc = "The armbands of a deadly martial artist. Makes you pretty keen to put an end to evil in an extremely violent manner."
+	icon_state = "rapid"
+	item_state = "rapid"
+	enhancement = 10 //omae wa mou shindeiru
+	var/warcry = "AT"
+	secondary_trait = TRAIT_NOHARDCRIT
+	
+/obj/item/clothing/gloves/fingerless/pugilist/rapid/Initialize()
+	. = ..()
+	ADD_TRAIT(src, TRAIT_NODROP, GLOVE_TRAIT)
+
+/obj/item/clothing/gloves/fingerless/pugilist/rapid/Touch(mob/living/target,proximity = TRUE)
+	if(!istype(target))
+		return
+
+	var/mob/living/M = loc
+	M.changeNext_move(CLICK_CD_RAPID)
+	if(warcry)
+		M.say("[warcry]", ignore_spam = TRUE, forced = "north star warcry")
+
+	.= FALSE
+
+/obj/item/clothing/gloves/fingerless/pugilist/rapid/attack_self(mob/user)
+	var/input = stripped_input(user,"What do you want your battlecry to be? Max length of 6 characters.", ,"", 7)
+	if(input)
+		warcry = input
+
+/obj/item/clothing/gloves/fingerless/pugilist/rapid/hug
+	name = "Hugs of the North Star"
+	desc = "The armbands of a humble friend. Makes you pretty keen to go let everyone know how much you appreciate them!"
+	warcry = "owo" //Shouldn't ever come into play
+	enhancement = 0
+	secondary_trait = TRAIT_PACIFISM //You are only here to hug and be friends!
+
+/obj/item/clothing/gloves/fingerless/pugilist/rapid/hug/Touch(mob/living/target,proximity = TRUE)
+	if(!istype(target))
+		return
+
+	var/mob/living/M = loc
+
+	if(M.a_intent == INTENT_HELP)
+		if(target.health >= 0 && !HAS_TRAIT(target, TRAIT_FAKEDEATH)) //Can't hug people who are dying/dead
+			if(target.on_fire || target.lying) //No spamming extinguishing, helping them up, or other non-hugging/patting help interactions
+				return
+			else
+				M.changeNext_move(CLICK_CD_RAPID)
+	. = FALSE
```

Comment:
Do early returns instead of `if` nesting
For example:
```
if(M.a_intent != INTENT_HELP)
	return FALSE
if(target.health < 0 || HAS_TRAIT(target, TRAIT_FAKEDEATH)) //Can't hug people who are dying/dead
	return FALSE
```
On the second check you could also use `if(stat != CONSCIOUS)` as well, depending on what you want.